src/sql/src/plan/hir.rs

```diff
@@ -131,7 +133,7 @@ pub enum HirRelationExpr {
         scalars: Vec<HirScalarExpr>,
     },
     CallTable {
-        func: TableFunc,
+        func: TableFuncMaybeWithOrdinality,
         exprs: Vec<HirScalarExpr>,
     },
```

Comment:
What's the downside of adding a `CallTableWithOrdinality` HIR node? Moving the variant into the `with_ordinality` field seems a bit odd from designing the enum, because it fundamentally changes how we evaluate the table functions.

Reply:
Unfortunately it's not possible to fix the problem just by doing a change at the HIR level. The fundamental problem is that MIR and LIR operators emit unordered collections, so we have to somehow execute `WITH ORDINALITY` inside `FlatMap`, which necessitates changes at all levels (HIR, MIR, optimizer transforms, LIR, rendering).

(Btw. I'm planning to refactor the PR a bit to not introduce the new `TableFuncMaybeWithOrdinality` struct that wraps `TableFunc`, but instead create a new enum variant of `TableFunc` for `WITH ORDINALITY`, which would wrap a normal `TableFunc`. And then the code that is currently in `TableFuncMaybeWithOrdinality` will move into `TableFunc`'s implementation. This is not a fundamental change in what the PR does, but I think this will make the PR's diff shorter.)